js/renovation/ui/grids/data_grid/utils/get_updated_options.ts

```diff
@@ -21,17 +21,17 @@ function compare(resultPaths: ResultItem[], item1, item2, key: string): void {
       resultPaths.push(getDiffItem(key, item2, item1));
     } else {
     // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      const diffPaths = objectDiffs(item1, item2);
+      const diffPaths = objectDiffs(item1, item2, fullPropName);
       resultPaths.push(...diffPaths.map((item) => ({ ...item, path: `${key}.${item.path}` })));
     }
   } else if (type1 === 'array') {
-    if (key !== 'columns' && item1 !== item2) {
+    if (key !== 'columns' && !(key === 'items' && fullPropName.includes('toolbar')) && item1 !== item2) {
```

Comment:
Why is this check necessary here? Where is jest tests for this change?

Reply:
This check is necessary. Before that function `compare` went deep comparing changes only if it's some of column options. In toolbar I also need to deeply check what option was changed. 

Without this check when I call `grid.option('toolbar.items[0].disabled', true)`, in `optionChanged` argument `fullName` would be `toolbar.items`, not `toolbar.items[0].disabled`